libsimulator/src/RoutingEngine.cpp

```diff
@@ -19,16 +19,131 @@
 #include <CGAL/Exact_predicates_exact_constructions_kernel.h>
 #include <CGAL/Polygon_with_holes_2.h>
 #include <CGAL/Surface_mesh.h>
+#include <CGAL/Triangulation_conformer_2.h>
 #include <CGAL/Triangulation_vertex_base_with_info_2.h>
 #include <CGAL/draw_triangulation_2.h>
 #include <CGAL/mark_domain_in_triangulation.h>
 
 #include <limits>
 #include <memory>
-#include <queue>
 #include <unordered_map>
 #include <vector>
 
+using Kernel = CGAL::Simple_cartesian<double>;
+using Point_2 = Kernel::Point_2;
+
+// for debuging
+void export_triangles(const CDT& cdt, const std::string& filename)
+{
+    std::ofstream out(filename);
+    if(!out) {
+        std::cerr << "Cannot open file " << filename << " for writing." << std::endl;
+        return;
+    }
+
+    for(auto fit = cdt.finite_faces_begin(); fit != cdt.finite_faces_end(); ++fit) {
+        if(!fit->get_in_domain())
+            continue;
+
+        for(int i = 0; i < 3; ++i) {
+            const auto& p = fit->vertex(i)->point();
+            out << CGAL::to_double(p.x()) << "," << CGAL::to_double(p.y()) << " ";
+        }
+        out << "\n";
+    }
+
+    std::cout << "Exported triangles to " << filename << std::endl;
+}
```

Comment:
Check the visualiser code how to do this in python, unless im mistaken there is no need to do this from C++.